merlion/models/deep_base.py

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright (c) 2022 salesforce.com, inc.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+# For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+#
+"""
+    Base class for Deep Learning Models
+"""
+
+import copy
+import logging
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from scipy.stats import norm
+from abc import abstractmethod
+
+try:
+    import torch
+    import torch.nn as nn
+except ImportError as e:
+    err = (
+        "Try installing Merlion with optional dependencies using `pip install salesforce-merlion[deep-learning]` or "
+        "`pip install `salesforce-merlion[all]`"
+    )
+    raise ImportError(str(e) + ". " + err)
+
+from merlion.models.base import Config, ModelBase
+from merlion.plot import Figure
+from merlion.transform.base import TransformBase, Identity
+from merlion.transform.factory import TransformFactory
+from merlion.utils.misc import initializer
+from merlion.utils.time_series import to_pd_datetime, to_timestamp, TimeSeries, AggregationPolicy, MissingValuePolicy
+
+logger = logging.getLogger(__name__)
+
+
+class DeepConfig(Config):
+    """
+    Config Object used to define a deep learning (pytorch) model
+    """
+
+    @initializer
+    def __init__(
+        self,
+        lr: float = 1e-3,
+        batch_size: int = 256,
+        num_epochs: int = 10,
+        optim_name: str = "adam",
+        criterion: str = "mse",
```

Comment:
Can you define enums which indicate the valid kinds of optimizers and loss functions? You could do something like
```python
class Optimizer(Enum):
    Adam = torch.optim.Adam
    AdamW = torch.optim.AdamW
    ...
```
You should define `DeepConfig.optimizer` as a property with an appropriate getter/setter (see [here](https://github.com/salesforce/Merlion/blob/main/merlion/models/anomaly/change_point/bocpd.py#L117) for an example). Then, you can create the optimizer using
```python
self.optimizer = self.config.optimizer.value(self.deep_model.parameters(), lr=self.config.lr)
```

Reply:
Additionally, can you rename `criterion` to `loss_fn`?